pkg/arvo/app/eth-watcher.hoon

```diff
@@ -1,571 +1,391 @@
-::  watcher: ethereum event log collector
+::  eth-watcher: ethereum event log collector
 ::
-/+  *eth-watcher
+/-  *eth-watcher
+/+  tapp, stdio, ethio
+=,  ethereum-types
+=,  able:jael
+=>  |%
+    +$  app-state
+      $:  %0
+          dogs=(map path watchdog)
+      ==
+    ::
+    +$  context  [=path dog=watchdog]
+    +$  watchdog
+      $:  config
+          =number:block
+          =pending-logs
+          =history
+          blocks=(list block)
+      ==
+    ::
+    ::  history: newest block first, oldest event first
+    +$  history       (list loglist)
+    +$  pending-logs  (map number:block loglist)
+    ::
+    +$  peek-data
+      [%atom =next-block=number:block]
+    +$  in-poke-data
+      $:  %eth-watcher-poke
+          poke
+      ==
+    +$  out-poke-data  ~
+    +$  in-peer-data   ~
+    +$  out-peer-data
+      $:  %eth-watcher-diff
+          diff
+      ==
+    ++  tapp
+      %:  ^tapp
+        app-state
+        peek-data
+        in-poke-data
+        out-poke-data
+        in-peer-data
+        out-peer-data
+      ==
+    ++  tapp-async  tapp-async:tapp
+    ++  stdio  (^stdio out-poke-data out-peer-data)
+    ++  ethio  (^ethio out-poke-data out-peer-data)
+    --
 ::
-=,  ethereum
-=,  rpc
+::  Async helpers
 ::
-|%
-++  state
-  $:  eyes=(map name eye)
-  ==
-::
-++  eye
-  $:  config
-      latest-block=@ud
-      filter-id=@ud
-      poll-timer=(unit @da)
-      snapshot
-      sap=history
-  ==
+=>  |%
```

Comment:
Do you think these should go in lib/ethio?

Reply:
Probably, yeah! I moved them over. May or may not want to do `|_  =url` instead of taking it as a separate argument for each arm, but I think this is fine for now at least.

(There's also a secondary task here of "figure out what to do with eth stdlib". We have multiple places in zuse for Ethereum types and functions, and there's even some overlap... Worth thinking about how to do cleanup here, especially now that jael doesn't explicitly depend on Ethereum anymore.)